### PR TITLE
Move Student QuizDetails to Core

### PR DESCRIPTION
**Pull request by szabinst**
Moves Quiz related screens from Student to Core so it can be reused in Horizon. 
Added a prefix to Teacher Quiz files.

[ignore-commit-lint]

## Checklist

- [x] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>One or more files are below the minimum test coverage 50%</td>
    </tr>
  </tbody>
</table>


Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.18% | 91.18% | -0%
Core/Core/Planner/CalendarEvent/Model/API/PostCalendarEventRequest.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarEvent/View/CustomFrequencyComponents/OccurrencesCountInputDialog.swift | 0% | 0% | 0%
Core/Core/Planner/CalendarEvent/Model/Helpers/RecurrenceRule+SelectionDescription.swift | 43.33% | 43.33% | 0%
Core/Core/Grades/View/CustomSwipeAction.swift | 0% | 0% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 901e4a661f4deb99042774f1dcb4802b6d133954
</p>

File: Core/Core/Quizzes/QuizDetails/Student/StudentQuizDetailsViewController.swift [hunk 1]
```diff
@@ -62,28 +61,28 @@ class QuizDetailsViewController: ScreenViewTrackableViewController, ColoredNavVi
         self?.update()
     }
 
-    static func create(
+    static public func create(
```
Comment (rh12) [inline Core/Core/Quizzes/QuizDetails/Student/StudentQuizDetailsViewController.swift:64]: nit:
```suggestion
    public static func create(
```

File: Core/Core/Quizzes/QuizDetails/Teacher/Model/QuizAttribute.swift
Comment (rh12) [inline Core/Core/Quizzes/QuizDetails/Teacher/Model/QuizAttribute.swift:1]: Would it make sense to prefix this one with `Teacher`, as well?

**Review by rh12 — APPROVED** [review]
QA + 1 (smoke tested quizes in both apps)

**Review by vargaat — APPROVED** [review]

**Review by rh12 — APPROVED** [review]

**Pull request merged**